docs/explainer.md

```diff
@@ -9,6 +9,96 @@ See [Discourse](https://discourse.wicg.io/t/filling-the-remaining-gap-between-we
 User agents will need to carefully consider when to make the Raw Sockets API available to web applications,
 and what UI will be shown to the user.
 
+
+
+### Threat
+
+MITM attackers may inject sockets API calls into a web page.
+
+### Mitigation
+
+The API will only available in secure contexts (HTTPS).
+
+
```

Comment:
There is still a threat where a third-party is injected via a first-party context, like via any `<script>` tag that loads a script from a CDN for instance. Then the third-party script could silently gain access to a socket once the user grants permission.

Reply:
Isn't there some mechanism that can restrict access to APIs from third-party scripts?

Reply:
I don't think that is a MITM attack.

If I understand correctly, third-party script loaded in `<script>` tags with [Subresource Integrity](https://w3c.github.io/webappsec-subresource-integrity/) has the same security properties as any external script a web developer might copy/paste into their project repo, or list in their npm dependencies or similar.

Use of unscrutinized third party script falls into a different class of threats, like disgruntled developer risk, that affect all powerful APIs and are not specific to the web platform.

Reply:
> Isn't there some mechanism that can restrict access to APIs from third-party scripts?
 
No, unfortunately. 

> Use of unscrutinized third party script falls into a different class of threats, like disgruntled developer risk, that affect all powerful APIs and are not specific to the web platform.

Right, I think this is where we get into the fundamental philosophical differences in our communities: Mozilla folks are hesitant to add anything to the platform that allow a "disgruntled developer" level attack. Why we have no BlueTooth, MIDI, etc. etc... so, in a sense they are specific to the web platform in as far as Mozilla (and Apple) have not allowed shipped these kinds of APIs for this reason. 

That's not to say it's a show stopper - but it gives us a point at which we can ask, "what's this thing trying to do?"